stdlib/Logging/src/ConsoleLogger.jl

```diff
@@ -116,8 +116,16 @@ function handle_message(logger::ConsoleLogger, level::LogLevel, message, _module
     end
 
     # Generate a text representation of the message and all key value pairs,
-    # split into lines.
-    msglines = [(indent=0, msg=l) for l in split(chomp(convert(String, string(message))::String), '\n')]
+    # split into lines.  This is specialised to improve type inferesence,
+    # and reduce the risk of resulting method invalidations.
+    msglines = if message isa Base.AnnotatedString
+        message = Base.AnnotatedString(String(message), Base.annotations(message))
+        NamedTuple{(:indent, :msg), Tuple{Int, SubString{<:Union{Base.AnnotatedString{String}, String}}}}[
```

Comment:
style nit, and it seems to always return a particular SubString type?
```suggestion
        @NamedTuple{indent::Int, msg::SubString{Base.AnnotatedString{String}}}[
```

IIUC, you had been meaning to combine this with the next code path, so it returned an array with these either element type:
```julia
        Union{@NamedTuple{indent::Int, msg::SubString{Base.AnnotatedString{String}}},
              @NamedTuple{indent::Int, msg::SubString{String}}}[
```
but I don't really see the point in pushing the Union down to that level here, as it just means the compiler has to union-split that on some inner loop later instead of happening here